CMakeLists.txt

```diff
@@ -293,6 +293,17 @@ add_subdirectory(arch)
 add_subdirectory(drivers)
 add_subdirectory(tests)
 
+set(syscall_macros_h ${ZEPHYR_BINARY_DIR}/include/generated/syscall_macros.h)
+
+add_custom_target(syscall_macros_h_target DEPENDS ${syscall_macros_h})
```

Comment:
why are DEPENDS and OUTPUT the same here?
shouldn't DEPENDS be the script and not the header it generates?

Reply:
What OUTPUT are you talking about? The one in add_custom_command()? It's different than the one in add_custom_target().

Reply:
aha, I think I am misreading this, yes I see it now